clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -817,8 +821,12 @@ static void DiagnoseHLSLRegisterAttribute(Sema &S, SourceLocation &ArgLoc,
       S.Diag(TheDecl->getLocation(),
              diag::warn_hlsl_user_defined_type_missing_member)
           << regTypeNum;
-
-    return;
+    // Space argument cannot be specified for global constants
+    if (!isDeclaredWithinCOrTBuffer(TheDecl)) {
```

Comment:
What's the `!isDeclaredWithinCOrTBuffer()` test for?  If I comment that out then the tests still pass.

Reply:
The diagnostic implies that space can only be specified on non-global constants. The reason that condition is needed is so that we ensure that the attribute is applied on a non-global decl. I've added 2 tests to exercise this behavior.